test/realm_test.dart

```diff
@@ -829,6 +829,172 @@ Future<void> main([List<String>? args]) async {
+    final isRealm1Cancelled = getRealmAsync(configuration, cancellationToken: cancellationToken1).thenIsCancelled();
+
+    var cancellationToken2 = CancellationToken();
+    final isRealm2Cancelled = getRealmAsync(configuration, cancellationToken: cancellationToken2).thenIsCancelled();
+    cancellationToken1.cancel();
+    expect(await isRealm1Cancelled, isTrue);
+    expect(await isRealm2Cancelled, isFalse);
+  });
+
+  baasTest('Realm.open (flexibleSync) - open two different Realms(for user1 and user2) and cancel only the second', (appConfiguration) async {
+    final app = App(appConfiguration);
+
+    final user1 = await app.logIn(Credentials.anonymous());
+    final configuration1 = Configuration.flexibleSync(user1, [Task.schema]);
+    var cancellationToken1 = CancellationToken();
+    final isRealm1Cancelled = getRealmAsync(configuration1, cancellationToken: cancellationToken1).thenIsCancelled();
+
+    final user2 = await app.logIn(Credentials.anonymous(reuseCredentials: false));
+    final configuration2 = Configuration.flexibleSync(user2, [Task.schema]);
+    var cancellationToken2 = CancellationToken();
+    final isRealm2Cancelled = getRealmAsync(configuration2, cancellationToken: cancellationToken2).thenIsCancelled();
+
+    cancellationToken2.cancel();
+    expect(await isRealm2Cancelled, isTrue);
+    expect(await isRealm1Cancelled, isFalse);
+  });
+
+  baasTest('Realm.open (flexibleSync) - cancel after realm is returned is no-op', (appConfiguration) async {
+    final app = App(appConfiguration);
+    final credentials = Credentials.anonymous();
+    final user = await app.logIn(credentials);
+    final configuration = Configuration.flexibleSync(user, [Task.schema]);
+
+    var cancellationToken = CancellationToken();
+    final realm = await getRealmAsync(configuration, cancellationToken: cancellationToken);
+
+    expect(realm, isNotNull);
+    expect(realm.isClosed, false);
+
+    cancellationToken.cancel();
+    expect(realm.isClosed, false);
+  });
+
+  baasTest('Realm.open (flexibleSync) - listen for download progress of an empty realm', (appConfiguration) async {
+    final app = App(appConfiguration);
+    final credentials = Credentials.anonymous();
+    final user = await app.logIn(credentials);
+    final configuration = Configuration.flexibleSync(user, [Task.schema]);
+
+    int printCount = 0;
+
+    var syncedRealm = await getRealmAsync(configuration, onProgressCallback: (syncProgress) {
+      print("PROGRESS: transferredBytes: ${syncProgress.transferredBytes}, totalBytes:${syncProgress.transferableBytes}");
+      printCount++;
+    });
+
+    expect(syncedRealm.isClosed, false);
+    expect(printCount, isNot(0));
+  });
+
+  baasTest('Realm.open (flexibleSync) - download a populated realm', (appConfiguration) async {
+    final app = App(appConfiguration);
+    final config = await _addDataToAtlas(app);
+    var syncedRealm = await getRealmAsync(config);
+    expect(syncedRealm.isClosed, false);
```

Comment:
This should verify the contents of the Realm, otherwise there's no guarantee that `Realm.open` actually downloads the latest state.